### DISCUSSION TITLE
No Router in Aqara Single Switch Module T1 with Neutral (DLKZMK11LM)

### DISCUSSION THREAD

**Discussion by SOz92** [Koenkk/zigbee2mqtt#18781] [Devices]
Hi, I have bought a couple of this device with neutral and I have integrated them into zigbee2mqtt Home assistant, using a Sonoff coordinator in its latest version, and it detects the device as End Device.

Having a neutral, shouldn't it detect it as a router and repeat the signal? I have read in another discussion that with version 20211217 this worked...

Has anyone had the same problem, or knows if it can be fixed?

Thanks!

**Comment by DzianisFrydliand** [comment]
Hi all,
Just paired the same device and having the same experience. Device class shown is EndDevice

**Comment by fox1047** [comment]
I bought those relays and have the same issue. My coordinator is Sonoff dongle E with ember firmware.

**Comment by fox1047** [comment]
After some trial and error I was able to pair both my relays as routers. 

1. I force removed relay from z2m, 
2. Started pairing on relay for it to forget my zigbee network
3. Removed power from relay
4. Restarted Z2M
5. Permitted joining on coordinator only
6. Connected relay to power and started pairing

I don't know which of those steps might be unnecessary, but at least it worked.

**Reply by SirSheik** [reply]
Your instructions worked perfectly. Thank you very much for sharing it. 😉

**Comment by DzianisFrydliand** [comment]
Thanks, folks! After reading your steps I just clicked on the Interview button in z2m and the device is now displayed as a router. No need to re-pair even.

**Comment by mnymb** [comment]
Hello All I have integrated my Aqara M2 Hub to the Home Assistant via Matter. I was then able to successfully add some of the Aqara devices (motion sensor, Door and Window switch via Aqara App and then could see these devices in the Matter integration in HA. I have then pared the DLKZMK11LM switch with the same M2 Hub in Aqara App and it all worked well,  but for some reason I can't see this switch as a device in my HA via the Matter Integration. Is this compatibility issue for this particular switch? any advice will be appreciated
thanks

**Reply by SirSheik** [reply]
Hello.
Are you using Zigbee2Mqtt to integrate the device? I'm sorry but I don't understand how your question is related to this project.

**Reply by mnymb** [reply]
Hi Thanks for your reply. I am trying to reach out to the folks who have experience in integration of the Aqara DLKZMK11LM in HA. As for your first question I'm not. what I actually try doing is integrating the DLKZMK11LM into the HA via Aqara M2 Hub which itself is integrated into the HA via Matter integration. 
I do apologize if I have used the wrong trail..........
![Aqara Matter Integrations](https://github.com/user-attachments/assets/91116249-b0b6-45ef-b843-dd1f3abf1146)
